test/test_serre_green_naghdi_1d.jl

```diff
@@ -0,0 +1,64 @@
+module TestSerreGreenNaghdi1D
+
+using Test
+using DispersiveShallowWater
+
+include("test_util.jl")
+
+EXAMPLES_DIR = joinpath(examples_dir(), "serre_green_naghdi_1d")
+
+@testset "SerreGreenNaghdiEquations1D" begin
+    @trixi_testset "serre_green_naghdi_soliton.jl" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "serre_green_naghdi_soliton.jl"),
+                            tspan=(0.0, 0.1),
+                            l2=[9.994998669268741e-7, 1.4703973445698635e-6, 0.0],
+                            linf=[6.5496216650196e-7, 1.027617322124641e-6, 0.0],
+                            cons_error=[0.0, 8.174581012099225e-10, 0.0],
+                            change_waterheight=0.0,
+                            change_entropy_modified=-3.1093350116861984e-11)
+
+        @test_allocations(semi, sol, allocs=550_000)
```

Comment:
Same question as before: Did you check how many allocation we actually have?

Reply:
I made them a bit stricter if possible, rounding them to the next reasonable multiple of `50_000`. For example, the first test has `523_304` allocations locally (rounded to `550_000`) and the second test has `393_360` allocations locally, rounded up to `450_000` (since it was too close to `400_000`).

Reply:
Sounds good.